### PR TITLE
Refine: 1.0.13

### PR DESCRIPTION
**Pull request by hanyulo**

### CODE REVIEW

File: src/components/side-bar.js [hunk 1]
```diff
@@ -15,20 +16,48 @@ for (const key in sectionStrings) {
...
+      if (tem != null) return tem.slice(1).join(' ').replace('OPR', 'Opera')
+    }
+    M = M[2] ? [M[1], M[2]] : [navigator.appName, navigator.appVersion, '-?']
+    tem = ua.match(/version\/(\d+)/i)
+    if (tem != null) {
+      M.splice(1, 1, tem[1])
+    }
+    return M.join(' ')
+  }())
+  return navigator.sayswho
```
Comment (nickhsine) [inline src/components/side-bar.js]: here is a easy way to detect desktop safari browser
https://stackoverflow.com/questions/7944460/detect-safari-browser

File: src/components/side-bar.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/no-did-mount-set-state */
```
Comment (nickhsine) [inline src/components/side-bar.js]: You can setState in ComponentWillMount.
Hence, you don't have to disable this eslint rule.

File: src/components/editor-picks-mobile.js [hunk 3]
```diff
@@ -118,6 +128,10 @@ class EditorPicksMobile extends SwipableMixin {
       )
     })
 
+    // <TRLink
+    //   href={href}
+    // >
+    //   </TRLink>
```
Comment (nickhsine) [inline src/components/editor-picks-mobile.js]: remove commented lines

File: src/components/editor-picks-mobile.js [hunk 4]
```diff
@@ -126,12 +140,8 @@ class EditorPicksMobile extends SwipableMixin {
...
         >
           <Category>{_.get(post, 'categories[0].name', '')}</Category>
-          <Title>
-            <TRLink
-              href={href}
-            >
-              {_.get(post, 'title', '')}
-            </TRLink>
+          <Title onClick={() => { this.redirect(href) }}>
+            {_.get(post, 'title', '')}
```
Comment (nickhsine) [inline src/components/editor-picks-mobile.js]: why could not you just use `TRLink` ?

Comment (nickhsine) [inline src/components/editor-picks-mobile.js]: We all forget a important thing; we have to check the style of the post.
If the post is `interactive` style, we need to use  `<a href="${full_url_to_post} target="_blank">` tag directly.

File: src/components/category-section.js [hunk 5]
```diff
@@ -31,60 +36,105 @@ const SectionWrapper = Section.extend`
...
+const MoreFrame = styled.div`
+  position: absolute;
+  bottom: -1px;
+  left: -1px;
+  width: 101%;
+  background-color: ${backgroundColor};
+  height: 36px;
+  text-align: center;
+`
+
```
Comment (nickhsine) [inline src/components/category-section.js]: why need this one?

Comment (hanyulo) [inline src/components/category-section.js]: MoreFrame work as a patch which has ${backgroundColor} to cover white background. The benefit is that I can still maintain a clean hover area. 

If I set styled component More to be patch, the hover area will become a long narrow rectangle.

File: src/components/side-bar.js (hunk 2 above)
Comment (hanyulo) [inline src/components/side-bar.js]: 👍

File: src/components/side-bar.js (hunk 1 above)
Comment (hanyulo) [inline src/components/side-bar.js]: 👍 💯

File: src/components/editor-picks-mobile.js (hunk 3 above)
Comment (hanyulo) [inline src/components/editor-picks-mobile.js]: 👍

File: src/components/category-section.js (hunk 5 above)
Comment (hanyulo) [inline src/components/category-section.js]: 👍

File: src/components/editor-picks-mobile.js [hunk 6]
```diff
@@ -140,8 +141,12 @@ class EditorPicksMobile extends SwipableMixin {
           isSelected={index === this.state.selected}
         >
           <Category>{_.get(post, 'categories[0].name', '')}</Category>
-          <Title onClick={() => { this.redirect(href) }}>
-            {_.get(post, 'title', '')}
+          <Title>
+            <TRLink
+              href={href}
```
Comment (nickhsine) [inline src/components/editor-picks-mobile.js]: `redirect={_.get(post, 'style')==='interactive'}`

Comment (hanyulo) [inline src/components/editor-picks-mobile.js]: 👍

File: src/components/editor-picks-mobile.js [hunk 7]
```diff
@@ -135,6 +137,7 @@ class EditorPicksMobile extends SwipableMixin {
 
     const textFrameContent = data.map((post, index) => {
       const href = `a/${_.get(post, 'slug', 'error')}`
```
Comment (nickhsine) [inline src/components/editor-picks-mobile.js]: change to `i/${href}` if `style === 'interactive'`

File: src/components/editor-picks.js [hunk 8]
```diff
@@ -166,6 +166,7 @@ class EditorPicks extends React.Component {
     const FlexItems = (() => {
       return data.map((obj, i) => {
         const href = `a/${_.get(obj, 'slug', 'error')}`
```
Comment (nickhsine) [inline src/components/editor-picks.js]: change to `i/${href}` if `style === 'interactive'`

File: src/components/latest.js [hunk 9]
```diff
@@ -131,11 +131,16 @@ const Title = styled.div`
 class Latest extends React.PureComponent {
   render() {
     const latestItems = this.props.data.map((item) => {
+      const href = `a/${_.get(item, 'slug', 'error')}`
```
Comment (nickhsine) [inline src/components/latest.js]: change to `i/${href}` if `style === 'interactive'`

File: src/components/infographic-section.js [hunk 10]
```diff
@@ -174,7 +178,7 @@ class Infographic extends React.PureComponent {
     const href = `https://www.twreporter.org/i/${slug}`
```
Comment (nickhsine) [inline src/components/infographic-section.js]: change to `i/${href}` if `style === 'interactive'`

**Comment by nickhsine** [comment]
I think you might need to check the `style` of post to decide the uri for every link of post.

**Comment by nickhsine** [comment]
:ok_man: :1st_place_medal: 💯

**Pull request merged**